common/Resources/ti.internal/extensions/js/Intl.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2020 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+/**
+ * Generates a "creation" properties dictionary for Titanium's Collator, DateTimeFormat, and NumberFormat proxies
+ * from the given Intl type's constructor arguments.
+ * @param {Object[]} args
+ * The arguments array that was passed into Intl Collator, DateTimeFormat, or NumberFormat type's constructor.
+ * @param {Function} supportedFormatLocalesFunction Reference to a supportedLocalesOf() function.
+ * @return {Object} Returns a properties dictionary to be passed into a Titanium proxy's constructor.
+ */
+function makeTiFormatCreationPropertiesFrom(args, supportedFormatLocalesFunction) {
+	const properties = {};
+	if (args.length >= 1) {
+		if (typeof args[0] === 'string') {
+			properties.locale = args[0];
+		} else if (Array.isArray(args[0])) {
+			const supportedLocales = supportedFormatLocalesFunction(args[0]);
+			if (supportedLocales.length > 0) {
+				properties.locale = supportedLocales[0];
+			}
+		} else if (typeof args[0] === 'object') {
+			properties.options = args[0];
+		}
+	}
+	if ((args.length >= 2) && !properties.options) {
+		if (typeof args[1] === 'object') {
+			properties.options = args[1];
+		}
+	}
+	return properties;
+}
+
+// Add "Intl" APIs missing on Android.
+if (OS_ANDROID) {
```

Comment:
<!--
  1 failure:  common/Resources/...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/js/Intl.js;
  Line: 39;
-->
- :no_entry_sign: common/Resources/ti.internal/extensions/js/Intl.js line 39 – 'OS_ANDROID' is not defined. (no-undef)